### PR TITLE
Outposts v0.2: Test PR

### PR DESCRIPTION
**Pull request by tmtmtl30**
## About The Pull Request

ships spawn in (non-exclusive, so don't get too comfortable) hangars at the outpost sized to them, connected to a new "main level" of the outpost via an elevator. new hangars are generated as necessary as ships spawn / dock. let me know if radio works across hangars or not

~~this is mega-unfinished and not fit for merge, but it IS fit for test, and that's what we do around here isn't it???~~

i'm basically done with this. that took a hot minute.
for detailed documentation, check here: https://hackmd.io/@Xegx-gsTR2WyYII99ZKtrg/B1updZdK2

## Why It's Good For The Game

it's cool

## Changelog

:cl:
add: Ships now spawn at the outpost, in hangars. Take the elevator in the top-right down to the main level.
add: 100% guaranteed most realistic ss13 elevator experience if you ignore all the things that aren't realistic
/:cl:

### CODE REVIEW

**Comment by tmtmtl30** [comment]
unit tests are angry but it works locally. my solution for getting ships to spawn on the outpost was a quick hack and i'll think of a better one when i'm not dead tired

**Comment by UltraBigRat** [comment]
overall pretty good, some feedback:
Maybe add ship names (and maybe classes) to the  elevator floor buttons since all floors other then the first one are a hangar for a single ship.
The elevator button on the outside should probably just call instead of moving it up or down, and maybe the elevator should only close upon it beginning to move not when people enter. 
overall pretty good though, just needs polish.

**Comment by tmtmtl30** [comment]
minor update: admins can now spawn ships in locations other than the outpost

**Review by spockye — DISMISSED** [review]
image
small decal error here. Try changing the layers around!

image
did you mean for them to be this faded?

image
WOAH HIGH TECH HOLOGRAPHIC COMPUTER SCREEN

image
use the table version of the wideband!

image
Button doesn't work

image
This button doesn't work

image
This button doesn't work

**Comment by goober3** [comment]
I don't know if this is a Mapping Issue or what but. a lot of Air Supply Pipes in the outpost are invisible in-game. odd!

**Comment by goober3** [comment]
will there bean option for ships to spawn in Space on ship configs? might be useful for more hazardous ships, like the Metis.

**Comment by tmtmtl30** [comment]
i don't have that planned but it'd definitely be doable code-wise

**Comment by tmtmtl30** [comment]
spockye has graced the PR with a second map for testing -- the map will be picked randomly at roundstart

**Comment by BarteG44** [comment]
the requirement for a kitchen was a chef holopad. the holopad is here but kitchen is not. i will resort to violence

**Comment by BarteG44** [comment]
Add moar holopadd got daym

**Comment by thgvr** [comment]
aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa

File: code/modules/overmap/objects/outpost/elevator/elevator_master.dm [hunk 1]
```diff
@@ -0,0 +1,368 @@
...
+	var/floor_move_time = 3 SECONDS
+	/// amount of time spent on a floor after arriving but before opening doors
+	var/door_open_time = 2 SECONDS
+	/// amount of time spent between opening doors and closing them
+	var/floor_idle_time = 7 SECONDS
+
+	/// List of floor datums that the elevator may move between.
+	var/list/datum/floor/floor_list = list()
+
+	/// The index into the floor list at which the
```
Comment (MarkSuckerberg) [inline code/modules/overmap/objects/outpost/elevator/elevator_master.dm]: the

File: html/changelogs/AutoChangeLog-pr-1812.yml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+author: MemedHams
+delete-after: true
+changes:
+  - bugfix: The rigger-c json is now correctly pathed to a .dmm file
```
Comment (MarkSuckerberg) [inline html/changelogs/AutoChangeLog-pr-1812.yml]: oop

**Review by MarkSuckerberg — CHANGES_REQUESTED** [review]
just two small non-functional things, otherwise it's legit from what I can tell

**Review by thgvr — APPROVED** [review]

**Review by MarkSuckerberg — APPROVED** [review]
lgtm

**Comment by goober3** [comment]
no freakin way lois

**Comment by BarteG44** [comment]
At last

**Pull request merged**